server/auth/types/openid/helper.ts

```diff
@@ -70,6 +71,18 @@ export async function callTokenEndpoint(tokenEndpoint: string, query: any): Prom
   };
 }
 
+export function composeLogoutUrl(
+  customLogoutUrl: string | undefined,
+  idpEndsessionEndpoint: string | undefined,
+  additionalQueryString: string
+) {
+  const logoutEndpont = customLogoutUrl || idpEndsessionEndpoint;
+  const logoutUrl = parseUrl(logoutEndpont!);
+  return logoutUrl.query
+    ? `${logoutEndpont}&${additionalQueryString}`
+    : `${logoutEndpont}?${additionalQueryString}`;
```

Comment:
I think it will fail into 'https://idp.com/path?a=b&&key=value' if given endpoint is 'https://idp.com/path?a=b&'. I would suggest using javascript built-in URL object which is more robust.

~~~
> function addPara(urlString) { 
    url = new URL(urlString);
    url.searchParams.append('key', 'value');
    return url.toString();
}
undefined
> addPara('https://idp.com/path?a=b')
"https://idp.com/path?a=b&key=value"
> addPara('https://idp.com/path?a=b&')
"https://idp.com/path?a=b&key=value"
> addPara('https://idp.com/path?a=b#')
"https://idp.com/path?a=b&key=value#"
~~~

Reply:
Agreed, updated accordingly